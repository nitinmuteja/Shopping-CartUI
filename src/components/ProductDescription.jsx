import React from 'react';
import Grid from "@material-ui/core/Grid";
import InputLabel from "@material-ui/core/InputLabel";
import NativeSelect from "@material-ui/core/NativeSelect";
import Button from "@material-ui/core/Button";
import {Box} from "@material-ui/core";

class ProductDescription extends React.Component {
    render() {

        return (
            <Grid container spacing={1}>
                <Grid item xs={12}>
                    <h2>Harry Potter and the Philosopher's Stone</h2>
                    <br/>
                    <Grid item xs={10}/>
                    <Grid item xs={2}>

                        by JK Rowling
                    </Grid>
                </Grid>

                <Grid item xs={12}>
                    Rating: 4
                </Grid>


                <Grid item xs={12}>
                    Price: 4
                </Grid>
                <Grid item xs={12}>
                    <h4>Book Description:</h4>
                    <p>
                        Hi, I'm Keshav, and my life is screwed. I hate my job and my girlfriend left me Hi, I'm Keshav,
                        and my life is screwed. I hate my job and my girlfriend left meHi, I'm Keshav, and my life is
                        screwed. I hate my job and my girlfriend left me
                    </p>
                </Grid>
                <Grid item xs={12}>
                    <InputLabel htmlFor="uncontrolled-native">Quantity</InputLabel>
                    <NativeSelect
                        defaultValue={10}
                        inputProps={{
                            name: 'name',
                            id: 'uncontrolled-native',
                        }}
                    >
                        <option value=""/>
                        <option value={10}>1</option>
                        <option value={20}>2</option>
                        <option value={30}>5</option>
                    </NativeSelect>
                </Grid>

                <Grid item xs={6} >
                    <Box pt={24}>
                    <Button variant="outlined" color="primary">
                        Quick Buy
                    </Button>
                    </Box>
                </Grid>
                <Grid item xs={6}/>



            </Grid>

        )


    }


}

export default ProductDescription;