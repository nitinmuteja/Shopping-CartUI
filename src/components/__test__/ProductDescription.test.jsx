import React from 'react';
import { render } from '@testing-library/react';
import ProductDescription from "../ProductDescription";




describe('Product Description Component tests', () => {


    it('should match snapshot', () => {
        const { container } = render(<ProductDescription/>);

        expect(container).toMatchSnapshot();
    });

});
