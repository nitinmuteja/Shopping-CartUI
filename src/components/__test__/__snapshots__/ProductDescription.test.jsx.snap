// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Description Component tests should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2>
        Harry Potter and the Philosopher's Stone
      </h2>
      <br />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        by JK Rowling
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      Rating: 4
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      Price: 4
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4>
        Book Description:
      </h4>
      <p>
        Hi, I'm Keshav, and my life is screwed. I hate my job and my girlfriend left me Hi, I'm Keshav, and my life is screwed. I hate my job and my girlfriend left meHi, I'm Keshav, and my life is screwed. I hate my job and my girlfriend left me
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
        for="uncontrolled-native"
      >
        Quantity
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline"
      >
        <select
          class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
          id="uncontrolled-native"
          name="name"
        >
          <option
            value=""
          />
          <option
            selected=""
            value="10"
          >
            1
          </option>
          <option
            value="20"
          >
            2
          </option>
          <option
            value="30"
          >
            5
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiNativeSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        class="MuiBox-root MuiBox-root-172"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Quick Buy
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    />
  </div>
</div>
`;
